crates/polars-core/src/chunked_array/ops/lambda.rs

```diff
@@ -473,12 +520,23 @@ impl LambdaExpression {
                     falsy.eval_slice(args)
                 }
             },
-            LambdaExpression::Length(expr) => match expr.eval_slice(args) {
-                AnyValue::Null => AnyValue::Null,
-                AnyValue::Binary(bytes) => AnyValue::Int32(bytes.len() as i32),
-                AnyValue::String(s) => AnyValue::Int32(s.len() as i32),
-                AnyValue::List(arr) => AnyValue::Int32(arr.len() as i32),
-                _ => AnyValue::Int32(1),
+            LambdaExpression::Length(expr) => {
+                match expr.eval_slice(args) {
+                    AnyValue::Null => AnyValue::Null,
+                    AnyValue::Binary(bytes) => {
+                        // case we have a binary slice, convert to utf8 as spark expects
+                        let new_str = from_utf8(bytes).unwrap();
```

Comment:
Did you test this with regular binary slices?

Reply:
Filter() will convert the String to Binary using as_binary().

However I will add array of binary as extra tests.

Reply:
Ike I wrote in chat, so far only int8/int16 had issues.